subsys/mgmt/mcumgr/smp_shell.c

```diff
@@ -176,13 +176,17 @@ static int smp_shell_tx_raw(const void *data, int len, void *arg)
 	return 0;
 }
 
-static int smp_shell_tx_pkt(struct zephyr_smp_transport *zst,
-			    struct net_buf *nb)
+static int smp_shell_tx_pkt(struct zephyr_smp_transport *zst, struct net_buf *nb)
 {
 	int rc;
 
-	rc = mcumgr_serial_tx_pkt(nb->data, nb->len, smp_shell_tx_raw, NULL);
-	mcumgr_buf_free(nb);
+	if (CONFIG_MCUMGR_SMP_SHELL_MTU == 0U) {
```

Comment:
A value of the `CONFIG_MCUMGR_SMP_SHELL_MTU` is known at compile time, why not just fail the compilation here?
Also, is the `MTU == 0` only MTU that is bad, because I am quite convinced that it is any MTU that can not handle minimal frame.

Reply:
This is outside the scope of this review and will need to be addressed later, the change above is the same behavior that mcumgr currently has but was moved out into the transport as per the request:
```
[de-nordic](https://github.com/de-nordic) [14 days ago](https://github.com/zephyrproject-rtos/zephyr/pull/49455#discussion_r955887120)

This can be done by zst_output callback.
```

Reply:
Yeah, what I meant was calling the mtu callback to modify how data is processed.
The check here is compile time check of 0 vs literal. This should just fail compilation.

Reply:
Fixed